### PR TITLE
Format the time string according to the precision of the time column

### PR DESCRIPTION
**Pull request by kamipo**
It is also necessary to format a time column like a datetime column.

### CODE REVIEW

File: activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb [hunk 1]
```diff
@@ -733,7 +733,7 @@ def register_integer_type(mapping, key, options) # :nodoc:
       end
 
       def extract_precision(sql_type)
-        if /datetime/ === sql_type
+        if /^(?:date)?time/ === sql_type
```
Comment (jeremy) [inline activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb]: Was the missing `^` anchor intentional before?

File: activerecord/test/cases/time_precision_test.rb [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 if ActiveRecord::Base.connection.supports_datetime_with_precision?
 class TimePrecisionTest < ActiveRecord::TestCase
+  self.use_transactional_fixtures = false
```
Comment (jeremy) [inline activerecord/test/cases/time_precision_test.rb]: Can `uses_transaction :test_formatting_time_according_to_precision` to skip the fixtures transaction for that single test.

Comment (kamipo) [inline activerecord/test/cases/time_precision_test.rb]: Ah, I was the first time to know `uses_transaction`. Thanks!

File: activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb (hunk 1 above)
Comment (kamipo) [inline activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb]: It seems the missing ^ anchor not intentional.

If you prefer to shortest pattern, it is possible to match `/time/`. It is only datetime, timestamp and time to match this in MySQL. These types can have a precision.

http://dev.mysql.com/doc/refman/5.6/en/fractional-seconds.html

File: Gemfile [hunk 3]
```diff
@@ -87,7 +87,7 @@ platforms :ruby do
   group :db do
     gem 'pg', '>= 0.18.0'
     gem 'mysql', '>= 2.9.0'
-    gem 'mysql2', '>= 0.3.13'
+    gem 'mysql2', '>= 0.3.18'
```
Comment (kamipo) [inline Gemfile]: 0.3.18 includes fix msec for time field is not passed. https://github.com/brianmario/mysql2/pull/579

**Pull request merged**